internal/config/validate.go

```diff
@@ -0,0 +1,72 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+package config
+
+import "fmt"
+
+func (nodeConfig *NodeConfig) Validate() {
```

Comment:
I see that as an improvement, like checking the URL format, if "block number" is an actual number and things alike. These were not being check before. So I propose it being a new PR/issue, since this is already a "long PR"